netcore/CMakeLists.txt

```diff
@@ -0,0 +1,19 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+project(netcore LANGUAGES NONE)
+
```

Comment:
Should we do a `find_program()` for `dotnet`?
Should we make building .NET Core bits optional with an `option()`?

Reply:
I can do these things. Should the build option be on by default?

Reply:
Maybe it should default to true if dotnet executable is found otherwise false? 🤷

Reply:
Although, given that this isn't nearly ready for release maybe it should be false.